From bd2a9753e5227c97bda737e00a00451d361449f3 Mon Sep 17 00:00:00 2001
From: Daiki Ueno <dueno@redhat.com>
Date: Mon, 24 Mar 2025 21:25:12 +0900
Subject: [PATCH] Fix build error with --with-ed25519-libsodium and
 --with-openssl

While libotcore can be configured with those options individually, the
latter is always required for OpenSSL's EVP functions. This splits the
ifdefs to accommodate that.

Signed-off-by: Daiki Ueno <dueno@redhat.com>
---
 src/libotcore/otcore.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/libotcore/otcore.h b/src/libotcore/otcore.h
index ceeb1a9265..3ce8f2a39f 100644
--- a/src/libotcore/otcore.h
+++ b/src/libotcore/otcore.h
@@ -25,7 +25,9 @@
 #ifdef HAVE_LIBSODIUM
 #include <sodium.h>
 #define USE_LIBSODIUM
-#elif defined(HAVE_OPENSSL)
+#endif
+
+#if defined(HAVE_OPENSSL)
 #include <openssl/evp.h>
 #include <openssl/x509.h>
 #define USE_OPENSSL
